plugins/packages/s3/lib/operations.ts

```diff
@@ -75,3 +75,9 @@ export async function signedUrlForPut(client: S3Client, options: object): Promis
   });
   return { url };
 }
+function uploadData(data: any, contentType: string) {
+  if (!data) {
+    return;
+  }
+  return typeof data === 'object' && contentType == 'application/json' ? JSON.stringify(data) : data;
```

Comment:
contentType can be `application/json; charset=utf-8`
shall we check if contentType includes `application/json`